templates/templates.go

```diff
@@ -55,29 +112,27 @@ func getEnvArray(envName string) ([]string, error) {
 	return strings.Split(env, ","), nil
 }
 
-func renderTemplate(templateText string) (templateResultBuffer bytes.Buffer, err error) {
-
-	// Create a FuncMap with which to register the function.
-	funcMap := template.FuncMap{
-		"getEnv": func(key string) (string, error) {
-			return getEnv(key)
-		},
-		"getEnvArray": func(key string) ([]string, error) {
-			return getEnvArray(key)
-		},
+func required(env interface{}) (string, error) {
+	if env == nil {
+		return "", errors.New("ENV variable is missing")
 	}
+	return env.(string), nil
```

Comment:
It is very error prone... It fails every time the `env` type is not a string :see_no_evil: 
https://go.dev/play/p/Evxdr3wWnGc
I know it won't happen here/now, but we never know how our code will change in the future, so it's always better to write code to not be prone to these kinds of bugs

Reply:
Agree, refactored based on your suggestion